### PR TITLE
Fix edge cases of getting attribute data from syntax

### LINKED ISSUE #75681: MarshallingAttributeAnalyzers throws InvalidCastException
### Description

NativeMarshallingAttributeAnalyzer and CustomMarshallerAttributeAnalyzer are throwing exceptions when building the runtime repo.

### Reproduction Steps

I upgraded an existing ADO pipeline to use .NET 7 RC 1 and it started to fail.
I have no idea what configuration in repo is relevant for repro, I'll be happy to share more details and link to pipeline as needed.

### Expected behavior

No errors

### Actual behavior

##[error]CSC(0,0): Error AD0001: Analyzer 'Microsoft.Interop.Analyzers.NativeMarshallingAttributeAnalyzer' threw an exception of type 'System.InvalidCastException' with message 'Unable to cast object of type 'Microsoft.CodeAnalysis.CSharp.Symbols.PublicModel.NonErrorNamedTypeSymbol' to type 'Microsoft.CodeAnalysis.IMethodSymbol'.'.
     7>CSC : error AD0001: Analyzer 'Microsoft.Interop.Analyzers.NativeMarshallingAttributeAnalyzer' threw an exception of type 'System.InvalidCastException' with message 'Unable to cast object of type 'Microsoft.CodeAnalysis.CSharp.Symbols.PublicModel.NonErrorNamedTypeSymbol' to type 'Microsoft.CodeAnalysis.IMethodSymbol'.'. [D:\a\_work\10\s\src\Product\UITester.Mac\Microsoft.Mashup.Client.UITester.Mac.csproj]
##[error]CSC(0,0): Error AD0001: Analyzer 'Microsoft.Interop.Analyzers.CustomMarshallerAttributeAnalyzer' threw an exception of type 'System.InvalidCastException' with message 'Unable to cast object of type 'Microsoft.CodeAnalysis.CSharp.Symbols.PublicModel.NonErrorNamedTypeSymbol' to type 'Microsoft.CodeAnalysis.IMethodSymbol'.'.
     7>CSC : error AD0001: Analyzer 'Microsoft.Interop.Analyzers.CustomMarshallerAttributeAnalyzer' threw an exception of type 'System.InvalidCastException' with message 'Unable to cast object of type 'Microsoft.CodeAnalysis.CSharp.Symbols.PublicModel.NonErrorNamedTypeSymbol' to type 'Microsoft.CodeAnalysis.IMethodSymbol'.'. [D:\a\_work\10\s\src\Product\UITester.Mac\Microsoft.Mashup.Client.UITester.Mac.csproj]

### Regression?

_No response_

### Known Workarounds

_No response_

### Configuration

_No response_

### Other information

Seems similar to https://github.com/dotnet/runtime/issues/72622 , but this time with InvalidCastException instead of InvalidOperationException.

### LINKED ISSUE #75706: LibraryImport analyzer crashes when encountering certain attributes
### Repro
1. git clone https://github.com/terrafx/terrafx.interop.windows
2. Open the solution in VS 2022 (17.4.0 Preview 2.0)
3. Build (and wait a minute or so)

### Expected Behavior

The analyzer works and reports any bugs.

### Actual Behavior

```
Severity	Code	Description	Project	File	Line	Suppression State	Detail Description
Error	AD0001	Analyzer 'Microsoft.Interop.Analyzers.CustomMarshallerAttributeAnalyzer' threw an exception of type 'System.InvalidOperationException' with message 'Sequence contains no elements'.	TerraFX.Interop.Windows (net6.0)	D:\Users\tagoo\source\repos\terrafx.interop.windows\sources\Interop\Windows\CSC	1	Active	Analyzer Failure
Error	AD0001	Analyzer 'Microsoft.Interop.Analyzers.NativeMarshallingAttributeAnalyzer' threw an exception of type 'System.InvalidOperationException' with message 'Sequence contains no elements'.	TerraFX.Interop.Windows (net6.0)	D:\Users\tagoo\source\repos\terrafx.interop.windows\sources\Interop\Windows\CSC	1	Active	Analyzer Failure
```

### PR DESCRIPTION
**Pull request by jkoritzinsky**
Fix two customer-discovered edge cases that were causing analyzer crashes.

Fixes #75681
Fixes #75706

### CODE REVIEW

**Review by tannergooding — APPROVED** [review]

File: src/libraries/System.Runtime.InteropServices/gen/LibraryImportGenerator/Analyzers/SyntaxExtensions.cs [hunk 1]
```diff
@@ -33,7 +33,17 @@ public static Location FindTypeExpressionOrNullLocation(this AttributeArgumentSy
                 switch (attributeTarget.Identifier.Kind())
                 {
                     case SyntaxKind.ReturnKeyword:
-                        return ((IMethodSymbol)targetSymbol).GetReturnTypeAttributes().First(attributeSyntaxLocationMatches);
+                        if (targetSymbol is IMethodSymbol method)
+                        {
+                            // Sometimes an attribute is put on a symbol that is nested within the containing symbol.
+                            // For example, the ContainingSymbol for an AttributeSyntax on a local function have a ContainingSymbol of the method.
```
Comment (stephentoub) [inline src/libraries/System.Runtime.InteropServices/gen/LibraryImportGenerator/Analyzers/SyntaxExtensions.cs:39]:
```suggestion
                            // For example, the ContainingSymbol for an AttributeSyntax on a local function has a ContainingSymbol of the method.
```

File: src/libraries/System.Runtime.InteropServices/gen/LibraryImportGenerator/Analyzers/SyntaxExtensions.cs [hunk 2]
```diff
@@ -43,7 +53,8 @@ public static Location FindTypeExpressionOrNullLocation(this AttributeArgumentSy
                 }
             }
             // Sometimes an attribute is put on a symbol that is nested within the containing symbol.
-            // For example, the ContainingSymbol for an AttributeSyntax on a parameter have a ContainingSymbol of the method.
+            // For example, the ContainingSymbol for an AttributeSyntax on a parameter have a ContainingSymbol of the method
+            // and an AttributeSyntax on a local function have a ContainingSymbol of the containing method.
```
Comment (stephentoub) [inline src/libraries/System.Runtime.InteropServices/gen/LibraryImportGenerator/Analyzers/SyntaxExtensions.cs:57]:
```suggestion
            // For example, the ContainingSymbol for an AttributeSyntax on a parameter has a ContainingSymbol of the method
            // and an AttributeSyntax on a local function has a ContainingSymbol of the containing method.
```

**Review by stephentoub — APPROVED** [review]

**Comment by jkoritzinsky** [comment]
Linux arm64 NativeAOT timeout is a known issue

**Pull request merged**